internal/provider/data_source_account.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/accounts"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAccount() *schema.Resource {
+	return &schema.Resource{
+		Description: "The boundary_account data source allows you to find a Boundary account.",
+		ReadContext: dataSourceAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the retrieved account.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			NameKey: {
+				Description:  "The name of the account to retrieve.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			DescriptionKey: {
+				Description: "The description of the retrieved account.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			AuthMethodIdKey: {
+				Description:  "The auth method ID that will be queried for the account.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func dataSourceAccountRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	md := meta.(*metaData)
+	opts := []accounts.Option{}
+
+	var name string
+	if v, ok := d.GetOk(NameKey); ok {
+		name = v.(string)
+	} else {
+		return diag.Errorf("no name provided")
+	}
+
+	var authMethodId string
+	if authMethodIdVal, ok := d.GetOk(AuthMethodIdKey); ok {
+		authMethodId = authMethodIdVal.(string)
+	} else {
+		return diag.Errorf("no auth method ID provided")
+	}
+
+	acl := accounts.NewClient(md.client)
+
+	als, err := acl.List(ctx, authMethodId, opts...)
```

Comment:
Could we pass in filter option (`withFilter`) here to reduce the amount of data we get back from Boundary. 
Eg. https://github.com/hashicorp/terraform-provider-boundary/blob/moduli-datasource-account/internal/provider/data_source_user.go#L111